Added release notes for 1.7 release (#365)
okteto/docs: src/content/self-hosted/install/releases.mdx

## 1.7.0
TBD

### Removal Notice
- Kubernetes [1.21](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.21.md) is no longer supported.

### Features

- Support for the annotation [dev.okteto.com/endpoints](cloud/ssl.mdx#customizing-the-endpoints-shown-in-the-okteto-ui) to customize the endpoints shown in the Okteto UI.
- Support for configuring the [service account](self-hosted/administration/configuration.mdx#buildkit) of buildkit.
- Support for [token-based authentication](self-hosted/administration/configuration.mdx#token) to simplify the installation of Okteto Self-Hosted.

> - [Personal Access Tokens](cloud/personal-access-tokens.mdx) now have expiration date. They expire after 180 days of their creation. For existing tokens, the expiration date is set to 180 days from when your instance is upgraded to this version

Reviewer: ```suggestion
- [Personal Access Tokens](cloud/personal-access-tokens.mdx) now have expiration date. They will automatically expire after 180 days. For existing tokens, the expiration date is set to 180 days from when your instance is upgraded to 1.7 or newer versions
```

> - Update the Okteto CLI version to [2.14.1](https://github.com/okteto/okteto/releases/tag/2.14.1)

Reviewer: ```suggestion
- Updated the Okteto CLI version to [2.14.1](https://github.com/okteto/okteto/releases/tag/2.14.1)
```

Can we group the dependency updates together?

- Allow to delete external resources individually from the UI
- Default installation of Okteto doesn't require an external storage for the registry configured. It uses filesystem storage by default

> - Default installation of Okteto doesn't require a PVC for [BuildKit](self-hosted/administration/configuration.mdx#token). It uses [filesystem storage](self-hosted/administration/configuration.mdx#filesystem) by default

Reviewer: Is this a breaking change? I had to make changes on my values.yaml when going from 1.6 to 1.7RC to keep it using volumes

Reviewer: Do we have docs on why you should use pvc vs file system?

Reviewer 2: > Is this a breaking change? I had to make changes on my values.yaml when going from 1.6 to 1.7RC to keep it using volumes 

I tested 1.7-rc.6 and the fix was working, wat rc was this happening on? I updated yesterday from 1.6 with PVC and then with the same config file to 1.7-rc6 and PVC was still being used

Author: @rberrelleza no changes are needed. You had to do changes because you hit a bug in the "migration" logic to keep using previous configuration. That bug was detected and fixed in the current release candidate. So, it shouldn't be a breaking change, we detect old configuration and we keep working as it was before.

> Do we have docs on why you should use pvc vs file system?

As far as I know, no, we don't have such documentation. I know Pablo had in mind to update some documentations about these new default values, but I don't see something like that

### Improvements

- When `buildkit.ingress.enabled` is true, BuildKit is exposed in the `buildkit.service.port` instead of port 1234. This makes easier to use other ingress controller.
- Optimized redeploy dialog experience. Infomation loads faster in the dialog, specially when the source is GitHub
- Improved confirmation dialogs messages

> - When your Okteto instance doesn't have an auth provider configured, the login screen shows a message with a link to the documentation about how to configure it

Reviewer: ```suggestion
- When your Okteto instance doesn't have an auth provider configured, the login screen shows a message with a link to the documentation on how to configure it
```

> - Upgrade ingress-nginx to [4.3.0](https://github.com/kubernetes/ingress-nginx/releases/tag/helm-chart-4.3.0)

Reviewer: ```suggestion
- Upgraded ingress-nginx to [4.3.0](https://github.com/kubernetes/ingress-nginx/releases/tag/helm-chart-4.3.0)
```

- Pod Security Policies are not installed if they are not available in the cluster

> - Moved `cloud.provider` helm chart value to [`registry.storage.provider`](self-hosted/administration/configuration.mdx#storage). We keep backward compatibility with values configured under `cloud.provider`

Reviewer: Can we mention when do we plan to deprecate the old secret name? 

Also, should this be in a separate part since we want users to take action a soon as possible? Maybe a deprecation notice?

Author: > Can we mention when do we plan to deprecate the old secret name?

Not sure we have plans to completely remove this. Probably in chart `2.0` but I'll sync with product about it

> Also, should this be in a separate part since we want users to take action a soon as possible? Maybe a deprecation notice?

I had doubts about this. I don't know how emphasize we want to make on this migration for users. Helm output will let you know about the new keys but it's probably a good idea to mention it also here

Reviewer: Having two different ways to accomplish the same thing makes it harder for our users to understand and for us to support them. If we don't have a clear plan to remove the old one, we should stick with the old one and not advertise the new one.  

I assume that we did the new one for a good reason, so I think the easier way is to have a clear deprecation path, and be transparent about it with our users.

Author: Added deprecation notice in this commit: https://github.com/okteto/docs/pull/365/commits/c8907ee6a9d2e0880686a4217cc90241d641011a

> - When using external storage for the registry, the secret with the credentials can be configured now in [`registry.secret`](self-hosted/administration/configuration.mdx#registry). We keep backward compatibility with `cloud.secret` helm value

Reviewer: I'm confused on how this is different from the one above.

Reviewer 2: One is regarding `cloud.provider` and the other is `cloud.secret` they are related but not the same.

just a proposal :

```suggestion
- When using external storage for the registry,  helm chart value `cloud.provider` has been moved to  [`registry.storage.provider`](self-hosted/administration/configuration.mdx#storage). We keep backward compatibility with values configured under `cloud.provider`
- The secret with the credentials for external storage provider for the registry should be configured now in [`registry.secret`](self-hosted/administration/configuration.mdx#registry). We keep backward compatibility with `cloud.secret`
```

Reviewer: ah I see, thanks!

> - When using external storage for the registry, the secret with the credentials can be configured now in [`registry.secret`](self-hosted/administration/configuration.mdx#registry). We keep backward compatibility with `cloud.secret` helm value

Reviewer 2: ```suggestion
- When using external storage for the registry, the secret with the credentials should be configured now in [`registry.secret`](self-hosted/administration/configuration.mdx#registry). We keep backward compatibility with `cloud.secret` helm value
```

### Bugfixes

- When redeploying a dev environment from the UI, there was a race condition that was preventing to use the right environment configuration (manifest, variables, branch, etc...)
- Sleeping resources now renders `No output.` logs instead of `Loading output...`
- Installer and destroy-all jobs are retried when there are transient errors during their execution

> - Avoid panics in the code when the a request to private endpoints doesn't have the cookie

Reviewer: Instead if this can we say "when the user is not authenticated" or something?

> - Don't let persistent namespaces to be slept from the UI

Reviewer: Is this by not displaying the button?

> - Fixed a problem when destroying dev environments from the UI when the manifest has defined an `image` property within the `destroy` section

Reviewer: What is the user facing impact?

## 1.7.0
TBD

### Removal Notice
- Kubernetes [1.21](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.21.md) is no longer supported.

### Deprecation Notice

`cloud.provider` and `cloud.secret` helm chart values are deprecated and they will be removed in the future 2.x release. Now, [`registry.storage.provider`](self-hosted/administration/configuration.mdx#storage) and [`registry.secret`](self-hosted/administration/configuration.mdx#registry) should be used instead as they configure settings for the registry external storage. This change is backward compatible and it will work with the old keys (if they are provided) until they are removed.

> So, imagine you have this configuration:

Author: @rberrelleza you mean add a section in this part mentioning the update to 1.7? https://deploy-preview-365--okteto-docs.netlify.app/docs/1.7/self-hosted/install/upgrade/

```yaml
cloud:
  secret:
    name: "okteto-cloud-secret"
    key: "key"
    token: "token"
  provider:
    aws:
      enabled: true
      region: us-west-2
      bucket: okteto-private-images
      iam:
        accessKeyID: "XXXXXX"
```

Now, it should be changed to and remove the config under `cloud`:

```yaml
registry:
  secret:
    name: "okteto-cloud-secret"
    accessKey: "key"
    secretKey: "token"
  storage:
    provider:
      aws:
        enabled: true
        region: us-west-2
        bucket: okteto-private-images
        iam:
          accessKeyID: "XXXXXX"
```

### Features

- Support for the annotation [dev.okteto.com/endpoints](cloud/ssl.mdx#customizing-the-endpoints-shown-in-the-okteto-ui) to customize the endpoints shown in the Okteto UI.
- Support for configuring the [service account](self-hosted/administration/configuration.mdx#buildkit) of buildkit.
- Support for [token-based authentication](self-hosted/administration/configuration.mdx#token) to simplify the installation of Okteto Self-Hosted.

> - [Personal Access Tokens](cloud/personal-access-tokens.mdx) now have expiration date. They will automatically expire after 180 days. For existing tokens, the expiration date is set to 180 days from when your instance is upgraded to 1.7 or newer versions

Reviewer: ```suggestion
- [Personal Access Tokens](cloud/personal-access-tokens.mdx) will automatically expire after 180 days. For existing tokens, the expiration date will be set to 180 days from when your Okteto instance is upgraded to 1.7 or newer versions
```

- Upgraded the Okteto CLI version to [2.14.1](https://github.com/okteto/okteto/releases/tag/2.14.1)
- Upgraded ingress-nginx to [4.3.0](https://github.com/kubernetes/ingress-nginx/releases/tag/helm-chart-4.3.0)

> - Allow to delete external resources individually from the UI

Reviewer: ```suggestion
- You can now delete external resources from the UI
```

> - Default installation of Okteto doesn't require an external storage for the registry configured. It uses filesystem storage by default

Reviewer: ```suggestion
- The default installation of Okteto doesn't require an external storage for the registry. It now uses filesystem storage by default
```

> - Default installation of Okteto doesn't require a PVC for [BuildKit](self-hosted/administration/configuration.mdx#token). It uses [filesystem storage](self-hosted/administration/configuration.mdx#filesystem) by default

Reviewer: Also,  I think the link to buildkit is wrong.

> - Default installation of Okteto doesn't require a PVC for [BuildKit](self-hosted/administration/configuration.mdx#token). It uses [filesystem storage](self-hosted/administration/configuration.mdx#filesystem) by default

Author: @rberrelleza you mean add a section in this part mentioning the update to `1.7`? https://deploy-preview-365--okteto-docs.netlify.app/docs/1.7/self-hosted/install/upgrade/

### Improvements

> - When `buildkit.ingress.enabled` is true, BuildKit is exposed in the `buildkit.service.port` instead of port 1234. This makes easier to use other ingress controller.

Reviewer: ```suggestion
- When `buildkit.ingress.enabled` is true, BuildKit is exposed in the `buildkit.service.port` instead of port 1234. This makes it easier to use your with other ingress controllers
```

- Optimized redeploy dialog experience. Infomation loads faster in the dialog, specially when the source is GitHub
- Improved confirmation dialogs messages
- When your Okteto instance doesn't have an auth provider configured, the login screen shows a message with a link to the documentation on how to configure it
- Pod Security Policies are not installed if they are not available in the cluster

> - Moved `cloud.provider` helm chart value to [`registry.storage.provider`](self-hosted/administration/configuration.mdx#storage). We keep backward compatibility with values configured under `cloud.provider`

Author: Yes, this is repeated from the deprecation notice, I will delete both as they are explained better above
